draccus/argparsing.py

```diff
@@ -203,7 +208,11 @@ def wrapper_outer(fn):
         def wrapper_inner(*args, **kwargs):
             argspec = inspect.getfullargspec(fn)
             argtype = argspec.annotations[argspec.args[0]]
-            cfg = parse(config_class=argtype, config_path=config_path, preferred_help=preferred_help)
+            if "config_dataclass" in kwargs and type(kwargs["config_dataclass"]) is argtype:
```

Comment:
i think i'd rather it be if the first argument is the config class that's in the signature of the function, do this.

Reply:
If I'm understanding you currectly, you just want the call to be `fn(config)` instead of `fn(config_dataclass=config)` ?

This was just a design choice, I was wondering what if it can lead to weird errors. Say if someone actually takes in a config object as intentional input? Eg:-
```
@draccus.wrap()
def main(cfg_base: TrainConfig, cfg_teacher:TrainConfig):
    print(cfg_base, cfg_teacher)


if __name__ == '__main__':
    config_teacher = TrainConfig(......)
    main(config_teacher)
 ```
 
 Curerntly this works, but if we just check the first argument against `argtype` then this might break?